org.spoofax.interpreter.core/src/main/java/org/spoofax/interpreter/library/ssl/SSL_immutable_map_from_list.java

```diff
@@ -31,15 +31,16 @@ protected SSL_immutable_map_from_list() {
                 return false;
             }
             final IStrategoTerm key = t.getSubterm(0);
+            final IStrategoTerm value = t.getSubterm(1);
             if(map.containsKey(key)) {
                 final IStrategoTerm oldValue = map.get(key);
-                env.setCurrent(f.makeTuple(oldValue, t.getSubterm(0)));
+                env.setCurrent(f.makeTuple(oldValue, value));
```

Comment:
This is `(oldValue, key)` in the original. Was that a bug?

Reply:
Yes, that was a bug. When you create a map from a list of pairs you get to merge values that have the same key. So the tuple passed to the merge strategy should be the two values.